server/middleware/bookVisitSessionValidator.ts

```diff
@@ -0,0 +1,100 @@
+import { RequestHandler, Response } from 'express'
+import createError from 'http-errors'
+import paths from '../constants/paths'
+import logger from '../../logger'
+
+const journeyOrder: string[] = [
+  paths.BOOK_VISIT.SELECT_PRISONER,
+  paths.BOOK_VISIT.SELECT_VISITORS,
+  paths.BOOK_VISIT.CHOOSE_TIME,
+  paths.BOOK_VISIT.ADDITIONAL_SUPPORT,
+  paths.BOOK_VISIT.MAIN_CONTACT,
+  paths.BOOK_VISIT.CHECK_DETAILS,
+  paths.BOOK_VISIT.BOOKED,
+]
+
+export default function bookVisitSessionValidator(): RequestHandler {
+  return (req, res, next) => {
+    const { booker, bookingJourney, bookingConfirmed } = req.session
+    const { baseUrl, method, path } = req
+    const requestPath = baseUrl + path
+
+    const journeyStage = journeyOrder.indexOf(requestPath)
+
+    // Unknown path
+    if (journeyStage === -1) {
+      return next(createError(404))
+    }
+
+    // No booker prisoner - any path
+    if (!booker.prisoners?.length) {
```

Comment:
Could you maybe combine all the conditions? Just seems 90% of them actually return the same thing?

Reply:
Hmm, yes, worth a look 🤔. We've got a couple of tickets next to add some more paths (for various drop-out points) so when I wedge those in I can see if it can be consolidated a bit.